lib/core/src/Cardano/Wallet/Primitive/Types/TokenPolicy/Gen.hs

```diff
@@ -32,7 +32,7 @@ genTokenNameSmallRange :: Gen TokenName
 genTokenNameSmallRange = elements tokenNamesSmallRange
 
 shrinkTokenNameSmallRange :: TokenName -> [TokenName]
-shrinkTokenNameSmallRange name = filter (< name) tokenNamesSmallRange
+shrinkTokenNameSmallRange _ = []
```

Comment:
Wouldn't it be better to have a shrinker like
```
if (x /= simplest) then [simplest] else []
```
to make counterexamples less noisy for human eyes?

Reply:
Yes. But is `AAAAAA` simpler than `CCCCCC` ? I hardly think so :upside_down_face:

Reply:
Right now, the generator produces names like `TokenA`, `TokenB`, `TokenC` etc ... And the only thing the shrinker does is to shrink lexicographically. So, `TokenC` is shrinked to [`TokenA`, `TokenB`] ... Same goes for the policy id and transaction ids.. This only takes needless CPU time and doesn't really make the output more readable. Hence their removal.

Reply:
Arguably, yes, there could be a one-step shrinker trying to shrink all things down to one "blessed" value. That should still be efficient enough and make the output a bit more uniform in case where there are many `TokenName` or `PolicyId` involved in the output :+1:

Reply:
Done :+1:

Reply:
Hi @KtorZ 

I'm not totally convinced these shrinkers are pointless, or even problematic.

The point here is not to make the output of _individual_ `TokenName` values more readable, but to generate simpler counterexamples where there are _multiple_ `TokenName` values involved (or multiple `TokenPolicyId` values).

This is useful in the case of tests for `TokenBundle`.

Let's say we have the following fully-shrunk counterexamples:
1. a counterexample with two `TokenName` values `x` and `y`, where both are different.
2. a counterexample with two `TokenName` values `x` and `y`, where both are the same.

In case no. 1, we know it's probably important for the values to be different for the test to fail.
In case no. 2, we know that it might be important for the values to be the same for the test to fail.

## Proposal

Instead of deleting these shrinkers, just don't call them in places where they're troublesome.

For example, in the QSM tests, we could just remove the reference to the shrinkers in the relevant `Arbitrary` instances.

So instead of writing:
```
instance Arbitrary TokenName where
    arbitrary = genTokenNameSmallRange
    shrink = shrinkTokenNameSmallRange
```

Just write:
```
instance Arbitrary TokenName where
    arbitrary = genTokenNameSmallRange
    -- No shrinking
```
 
Or if it's embedded in larger structure, such as a `TokenMap`, then write:
```
instance Arbitrary TokenMap where
    arbitrary = genTokenMapSmallRange
    -- No shrinking
```

Reply:
Please, see the updated commit :+1:

Reply:
> We could write the following instead:

Rather than completely removing shrinking for the state machine tests, it would be better to keep `if (x /= simplest) then [simplest] else []`.

Otherwise, I agree with @jonathanknowles's sentiment above.

Reply:
> This is useful in the case of tests for TokenBundle (i.e., in TokenBundleSpec.)

Didn't think of this 👍